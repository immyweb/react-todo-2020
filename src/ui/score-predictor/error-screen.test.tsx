import * as React from 'react';
import { shallow } from 'enzyme';
import ErrorScreen from './error-screen';

describe('<ErrorScreen />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ErrorScreen />);
    expect(wrapper).toMatchSnapshot();
  });
});
