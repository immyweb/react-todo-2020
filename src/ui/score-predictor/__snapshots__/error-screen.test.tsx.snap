// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorScreen /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <glamorous(img)
          src="https://www.thesun.co.uk/wp-content/uploads/2018/03/banana.png"
        />
        <glamorous(glamorous(h3))
          fontColour="#459977"
        >
          UH OH!
        </glamorous(glamorous(h3))>
        <glamorous(glamorous(h3))>
          THERE WAS A LOADING ERROR
        </glamorous(glamorous(h3))>
        <glamorous(button)
          background="#459977"
          onClick={[Function]}
        >
          Refresh page
        </glamorous(button)>
      </div>,
      "errorScreen": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <glamorous(img)
            src="https://www.thesun.co.uk/wp-content/uploads/2018/03/banana.png"
          />,
          <glamorous(glamorous(h3))
            fontColour="#459977"
          >
            UH OH!
          </glamorous(glamorous(h3))>,
          <glamorous(glamorous(h3))>
            THERE WAS A LOADING ERROR
          </glamorous(glamorous(h3))>,
          <glamorous(button)
            background="#459977"
            onClick={[Function]}
          >
            Refresh page
          </glamorous(button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": "https://www.thesun.co.uk/wp-content/uploads/2018/03/banana.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "UH OH!",
            "fontColour": "#459977",
          },
          "ref": null,
          "rendered": "UH OH!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "THERE WAS A LOADING ERROR",
          },
          "ref": null,
          "rendered": "THERE WAS A LOADING ERROR",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "background": "#459977",
            "children": "Refresh page",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Refresh page",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <glamorous(img)
            src="https://www.thesun.co.uk/wp-content/uploads/2018/03/banana.png"
          />
          <glamorous(glamorous(h3))
            fontColour="#459977"
          >
            UH OH!
          </glamorous(glamorous(h3))>
          <glamorous(glamorous(h3))>
            THERE WAS A LOADING ERROR
          </glamorous(glamorous(h3))>
          <glamorous(button)
            background="#459977"
            onClick={[Function]}
          >
            Refresh page
          </glamorous(button)>
        </div>,
        "errorScreen": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <glamorous(img)
              src="https://www.thesun.co.uk/wp-content/uploads/2018/03/banana.png"
            />,
            <glamorous(glamorous(h3))
              fontColour="#459977"
            >
              UH OH!
            </glamorous(glamorous(h3))>,
            <glamorous(glamorous(h3))>
              THERE WAS A LOADING ERROR
            </glamorous(glamorous(h3))>,
            <glamorous(button)
              background="#459977"
              onClick={[Function]}
            >
              Refresh page
            </glamorous(button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": "https://www.thesun.co.uk/wp-content/uploads/2018/03/banana.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "UH OH!",
              "fontColour": "#459977",
            },
            "ref": null,
            "rendered": "UH OH!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "THERE WAS A LOADING ERROR",
            },
            "ref": null,
            "rendered": "THERE WAS A LOADING ERROR",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "background": "#459977",
              "children": "Refresh page",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Refresh page",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
